lib/routes/password.js

```diff
@@ -29,13 +29,239 @@ module.exports = function (
   config
   ) {
 
+  const MIN_LAST_AUTH_PASSWORD_CHANGE = config.changePassword && config.changePassword.minLastAuth || 300000 // Default to 5 mins
+
+  const totpUtils = require('../../lib/routes/utils/totp')(log, config, db)
+
   function failVerifyAttempt(passwordForgotToken) {
     return (passwordForgotToken.failAttempt()) ?
       db.deletePasswordForgotToken(passwordForgotToken) :
       db.updatePasswordForgotToken(passwordForgotToken)
   }
 
   var routes = [
+    {
+      method: 'POST',
+      path: '/password/change',
```

Comment:
As a general nit, it feels like there is a lot of duplicated code between this route and `/password/change/finish`.  Is there a way to refactor things so they can share the bulk of the code here?